instrumentation/spring/spring-webflux-5.0/library/src/main/java/io/opentelemetry/instrumentation/spring/webflux/client/TraceWebClientSubscriber.java

```diff
@@ -44,25 +47,25 @@ public void onSubscribe(Subscription subscription) {
 
   @Override
   public void onNext(ClientResponse response) {
-    try (Scope ignored = otelContext.makeCurrent()) {
+    try (Scope ignored = otelParentContext.makeCurrent()) {
       this.actual.onNext(response);
     } finally {
-      instrumenter.end(otelContext, request, response, null);
+      instrumenter.end(otelClientContext, request, response, null);
```

Comment:
if we are making the parent context current for the downstream `onNext`, should we end the client span before calling downstream `onNext`?

Reply:
👍 

Also, FYI: I changed that to make our callback tests pass - turns out the old version was not passing the "correct" context, and it somehow worked for the javaagent -- most likely because of the underlying reactor-netty instrumentation.